modules/api-import-export/src/main/java/apim/restful/importexport/utils/APIExportUtil.java

```diff
@@ -425,66 +429,115 @@ public static void exportSequences(API api, APIIdentifier apiIdentifier, int ten
+                    OMElement seqElment = APIUtil.buildOMElement(sequence.getContentStream());
+                    if (sequenceName.equals(seqElment.getAttributeValue(new QName("name")))) {
+                        String sequenceFileName = sequence.getPath().substring(sequence.getPath().lastIndexOf('/'));
+                        sequenceDetails = new AbstractMap.SimpleEntry<String, OMElement>(sequenceFileName, seqElment);
+                        return sequenceDetails;
+                    }
+                }
             }
+
+        } catch (RegistryException e) {
+            log.error("Error while retrieving sequence from the registry " + e.getMessage());
+            throw new APIExportException("Error while retrieving sequence from the registry", e);
+        } catch (Exception e ) { //APIUtil.buildOMElement() throws a generic exception
+            log.error("Error while reading sequence content " + e.getMessage());
+            throw new APIExportException("Error while reading sequence content", e);
         }
+
+        return null;
+
     }
 
     /**
      * Store custom sequences in the archive directory
      *
-     * @param sequenceConfig Sequence configuration
-     * @param sequenceName   Sequence name
+     * @param sequenceDetails   Details of the sequence
      * @param direction      Direction of the sequence "in", "out" or "fault"
      * @param apiIdentifier  ID of the requesting API
      * @throws APIExportException If an error occurs while serializing XML stream or storing in
      *                            archive directory
      */
-    public static void writeSequenceToFile(OMElement sequenceConfig, String sequenceName, String direction,
-            APIIdentifier apiIdentifier) throws APIExportException {
+    private static void writeSequenceToFile(AbstractMap.SimpleEntry<String, OMElement> sequenceDetails, String direction,
+        APIIdentifier apiIdentifier) throws APIExportException {
         OutputStream outputStream = null;
         String archivePath = archiveBasePath.concat(File.separator + apiIdentifier.getApiName() + "-" +
                 apiIdentifier.getVersion()) + File.separator + "Sequences" + File.separator;
 
         String pathToExportedSequence = archivePath + direction + "-sequence" + File.separator;
-
-        String exportedSequenceFile = pathToExportedSequence + sequenceName + ".xml";
+        String sequenceFileName = sequenceDetails.getKey();
+        OMElement sequenceConfig = sequenceDetails.getValue();
+        String exportedSequenceFile = pathToExportedSequence + sequenceFileName;
         try {
             createDirectory(pathToExportedSequence);
             outputStream = new FileOutputStream(exportedSequenceFile);
             sequenceConfig.serialize(outputStream);
 
             if (log.isDebugEnabled()) {
-                log.debug(sequenceName + " retrieved successfully");
+                log.debug("Sequence retrieved successfully");
```

Comment:
It would be better to have the name/filename of the sequence in the log message.